pkg/apis/k0s.k0sproject.io/v1beta1/kuberouter.go

```diff
@@ -24,19 +24,32 @@ type KubeRouter struct {
 	MTU int `json:"mtu"`
 	// Kube-router metrics server port. Set to 0 to disable metrics  (default: 8080)
 	MetricsPort int `json:"metricsPort"`
-	// Activate Hairpin Mode (allow a Pod behind a Service to communicate to its own ClusterIP:Port)
-	HairpinMode bool `json:"hairpinMode"`
+	// Admits three values: "enabled" enables it globaly, "allowed" allows but services must be annotated explictly and "disabled"
+	Hairpin Hairpin `json:"hairpin"`
+	// DEPRECATED: Use hairpining instead. Activates Hairpin Mode (allow a Pod behind a Service to communicate to its own ClusterIP:Port)
+	HairpinMode bool `json:"hairpinMode,omitempty"`
```

Comment:
It's a pity that the better name is already taken by the old boolean setting. I wonder if we could merge those into one. Either accepting booeans or strings, just like apimachinery's `IntOrString` does it for ints and strings.

In any case I think we should rename both the type and the field name to `HairpinMode`, but just leave the json annotations untouched.

```suggestion
	// Admits three values: "Enabled" enables it globally, "Allowed" allows but services must be annotated explicitly and "Disabled"
	// +optional
	HairpinMode HairpinMode `json:"hairpin,omitempty"`
	// DEPRECATED: Use hairpining instead. Activates Hairpin Mode (allow a Pod behind a Service to communicate to its own ClusterIP:Port)
	HairpinEnabled bool `json:"hairpinMode,omitempty"`
```

Reply:
> Either accepting booleans or strings, just like apimachinery's IntOrString does it for ints and strings.
I like this idea very much